packages/puppeteer-core/src/bidi/core/UserContext.ts

```diff
@@ -196,6 +197,26 @@ export class UserContext extends EventEmitter<{
     return cookies;
   }
 
+  @throwIfDisposed<UserContext>(context => {
+    // SAFETY: Disposal implies this exists.
+    return context.#reason!;
+  })
+  async deleteCookie(...cookies: DeleteCookiesRequest[]): Promise<void> {
+    await Promise.all(cookies.map(async deleteCookieRequest => {
+      const filter = {
+        ...deleteCookieRequest
+      };
+      await this.#session.send('storage.deleteCookies', {
+        filter: filter,
+        partition: {
+          type: 'storageKey',
```

Comment:
should we delete using the type=context instead?

Reply:
moving it to the BrowseringContext.ts correspondingly

Reply:
Switched to `BrowsingContextPartitionDescriptor` and added tests.